### PR TITLE
Document query limitations for Materialize

### PR DESCRIPTION
**Pull request by mattlord**
v19 page: https://deploy-preview-1653--vitess.netlify.app/docs/19.0/reference/vreplication/materialize/

### CODE REVIEW

**Comment by mattlord** [comment]
@derekperkins and @rohit-nayak-ps whenever you have time, please take a look and let me know if you feel I'm missing anything. Once we're all happy with the content I will copy it to the previous release docs.

**Review by rohit-nayak-ps — APPROVED** [review]

File: content/en/docs/19.0/reference/vreplication/materialize.md [hunk 1]
```diff
@@ -10,6 +10,14 @@ weight: 40
...
 
 You can specify multiple tables to materialize using the [`create`](../../programs/vtctldclient/vtctldclient_materialize/vtctldclient_materialize_create/) sub-command's `--table-settings` flag.
+There are limitations, however, to the queries which can be used with `Materialize`:
+  - The query must be a `SELECT` statement
+  - Only the following operators are supported: `=`, `<`, `<=`, `>`, `>=`, `<>`, `!=` (e.g. no `IN`, `OR`, or `LIKE`)
+  - The query must be against a single table (so no `JOIN`s)
+  - The query cannot use `DISTINCT`
+  - The query cannot use a derived table
+  - Expressions in the query must have an alias, e.g. `select hour(c1) as c1_hour from t1`
+  - The `GROUP BY` expression cannot reference an aggregate expression such as `MAX` or `COUNT`
```
Comment (derekperkins) [inline content/en/docs/19.0/reference/vreplication/materialize.md:20]: We're using `COUNT(*)` in Materialize for our real-time billing usage


              "rules": [
                {
                  "match": "billing__usage_by_workspace",
                  "filter": "select workspace_id, date(from_unixtime(requested)) as requested_date, date(created_at) as created_date, 1 as type, count(*) as usage from workspaces_rankings__pulls group by workspace_id, requested_date, created_date, type"
                },
                {
                  "match": "billing__usage_by_project",
                  "filter": "select workspace_id, team_id as project_id, date(from_unixtime(requested)) as requested_date, date(created_at) as created_date, 1 as type, count(*) as usage from workspaces_rankings__pulls_by_team group by workspace_id, project_id, requested_date, created_date, type"
                }
              ]

Comment (mattlord) [reply content/en/docs/19.0/reference/vreplication/materialize.md:20]: Did you think something specific was wrong? The related limitation noted is that this count expression can't be referenced in a group by clause.

Comment (derekperkins) [reply content/en/docs/19.0/reference/vreplication/materialize.md:20]: Oh, my bad, I misread it as saying no aggregates

**Pull request merged**